### DISCUSSION TITLE
Remote checkout fails, but clone works

### DISCUSSION THREAD

**Discussion by allwalte** [community/community#56795] [Enterprise]
### Select Topic Area

Question

### Body

I'm trying to build an action that checks out both a repo from our GHE (the repo the workflow is in), as well as a public repo on github.com.  For the latter, I'm using the "github-server-url" argument, which seems to be taking effect because in the initialization phase for this second checkout I see ```remote add origin https://github.com/<owner>/<repo>```  (redactions mine, they're correct and visible in the pipeline :-) ).  However, the checkout fails at the ```Retrieving the default branch name``` step, with "Bad credentials".  I'm not specifying a PAT, it's my understanding that this repo is publicly accessible so I shouldn't need one.  I can certainly view everything from a private browser, not logged in.  And also, in the body of the job, if I do a ```git clone``` of that repo, it works.  So I'm not certain what's missing to make the checkout step work?

**Comment by airtower-luna** [comment]
If you *don't* specify a token, `actions/checkout` will use the `GITHUB_TOKEN` by default, and of course a token from your GHE instance won't be valid for `github.com`. I'm not sure how to tell it not to send an `Authorization` header at all, though, never had that problem. Maybe explicitly setting the token as empty? :thinking:

**Reply by allwalte** [reply]
Hmmm, interesting - empty token gives me ```"Input required and not supplied: token"```.  I see what you're saying though, and yeah, I can see it's saying authorization:basic in the initialization.  Will see if I can find anything searching with that in mind, anyway - thanks!

**Reply by allwalte** [reply]
As a follow-up, I tried ```persist-credentials: false``` but that did not prevent the header from being sent.

**Reply by allwalte** [reply]
Found [this doc](https://github.threatbuild.com/actions/checkout/blob/2541b1294d2704b0964813337f33b291d3f8596b/adrs/0153-checkout-v2.md#pat) from the scoping of checkout@v2 (using 3, but...this was what I could find:
```
When using the ${{github.token}} or a PAT, the token will be persisted in the local git config. The config key http.https://github.com/.extraheader enables an auth header to be specified on all authenticated commands AUTHORIZATION: basic <BASE64_U:P>.

Note:

    The auth header is scoped to all of github http.https://github.com/.extraheader
        Additional public remotes also just work.
```

"also just work" 😅

**Comment by mdmannan1599** [comment]
> ### Select Topic Area
> 
> Question
> 
> ### Body
> 
> I'm trying to build an action that checks out both a repo from our GHE (the repo the workflow is in), as well as a public repo on github.com.  For the latter, I'm using the "github-server-url" argument, which seems to be taking effect because in the initialization phase for this second checkout I see ```remote add origin https://github.com/<owner>/<repo>```  (redactions mine, they're correct and visible in the pipeline :-) ).  However, the checkout fails at the ```Retrieving the default branch name``` step, with "Bad credentials".  I'm not specifying a PAT, it's my understanding that this repo is publicly accessible so I shouldn't need one.  I can certainly view everything from a private browser, not logged in.  And also, in the body of the job, if I do a ```git clone``` of that repo, it works.  So I'm not certain what's missing to make the checkout step work?

How cam keep remote control app in my Samsung galaxy a10s device inside
